packages/@n8n/db/src/migrations/postgresdb/1764669632000-ChangeWorkflowStatisticsFKToNoAction.ts

```diff
@@ -0,0 +1,107 @@
+import type { MigrationContext, ReversibleMigration } from '../migration-types';
+
+export class ChangeWorkflowStatisticsFKToNoAction1764669632000 implements ReversibleMigration {
+	async up({ queryRunner, tablePrefix }: MigrationContext) {
+		// Step 1: Drop existing foreign key constraint and primary key
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics DROP CONSTRAINT IF EXISTS "fk_${tablePrefix}workflow_statistics_workflow_id"`,
+		);
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics DROP CONSTRAINT IF EXISTS "${tablePrefix}workflow_statistics_workflowId_fkey"`,
+		);
+		// Primary key constraint name varies: pk_ (from MigrateIntegerKeysToString) or _pkey (auto-generated)
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics DROP CONSTRAINT IF EXISTS "pk_${tablePrefix}workflow_statistics"`,
+		);
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics DROP CONSTRAINT IF EXISTS "${tablePrefix}workflow_statistics_pkey"`,
+		);
+
+		// Step 2: Add a new auto-generated id column as primary key
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics ADD COLUMN "id" SERIAL PRIMARY KEY`,
+		);
+
+		// Step 3: Make workflowId column nullable
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics ALTER COLUMN "workflowId" DROP NOT NULL`,
+		);
+
+		// Step 4: Add workflowName column to preserve workflow identity after deletion
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}workflow_statistics ADD COLUMN "workflowName" VARCHAR(128)`,
+		);
+
+		// Step 5: Populate workflowName from existing workflows
+		await queryRunner.query(
+			`UPDATE ${tablePrefix}workflow_statistics ws SET "workflowName" = we."name" FROM ${tablePrefix}workflow_entity we WHERE ws."workflowId" = we."id"`,
+		);
+
+		// Step 6: Add unique constraint on (workflowId, name) to maintain data integrity
+		// Note: Must be a regular unique index (not partial) for ON CONFLICT to work in upsertWorkflowStatistics
+		await queryRunner.query(
+			`CREATE UNIQUE INDEX "IDX_${tablePrefix}workflow_statistics_workflow_name" ON ${tablePrefix}workflow_statistics ("workflowId", "name")`,
```

Comment:
Do we need to make sure that this condition is true in all possible cases? For example if a workflow is deleted, the workflowId becomes null, creating a new workflow with the same name, could this be a conflict in this table?